vaadin-icons-flow-parent/vaadin-icons-flow/src/main/java/com/vaadin/flow/component/icon/AbstractIcon.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright 2000-2023 Vaadin Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.vaadin.flow.component.icon;
+
+import com.vaadin.flow.component.ClickNotifier;
+import com.vaadin.flow.component.Component;
+import com.vaadin.flow.component.HasStyle;
+import com.vaadin.flow.component.Tag;
+import com.vaadin.flow.component.dependency.JsModule;
+import com.vaadin.flow.component.dependency.NpmPackage;
+import com.vaadin.flow.component.shared.HasTooltip;
+import com.vaadin.flow.dom.ElementConstants;
+
+/**
+ * Component for displaying an icon from the
+ * <a href="https://vaadin.com/icons">Vaadin Icons</a> collection.
+ *
+ * @author Vaadin Ltd
+ * @see VaadinIcon
+ */
+@Tag("vaadin-icon")
+@NpmPackage(value = "@vaadin/polymer-legacy-adapter", version = "24.2.0-alpha11")
+@JsModule("@vaadin/polymer-legacy-adapter/style-modules.js")
+@NpmPackage(value = "@vaadin/icon", version = "24.2.0-alpha11")
+@JsModule("@vaadin/icon/src/vaadin-icon.js")
+public abstract class AbstractIcon extends Component
+        implements HasStyle, ClickNotifier<Icon>, HasTooltip {
```

Comment:
<img width="939" alt="Screenshot 2023-08-28 at 16 10 31" src="https://github.com/vaadin/flow-components/assets/1222264/de920c83-e350-4562-91aa-9e577ee0aa25">

HasStyle is redundant